lib/nerves_hub_cli.ex

```diff
@@ -16,14 +16,41 @@ defmodule NervesHubCLI do
     to_string(hostname)
   end
 
-  def home_dir do
-    override_dir =
-      Application.get_env(:nerves_hub_cli, :home_dir) || System.get_env("NERVES_HUB_HOME")
-
-    if override_dir == nil or override_dir == "" do
-      Path.expand("~/.nerves-hub")
-    else
-      override_dir
+  def home_dir() do
+    from_config = Application.get_env(:nerves_hub_cli, :home_dir)
+    from_env = System.get_env("NERVES_HUB_HOME")
+    xdg_home_nh = :filename.basedir(:user_data, "nerves-hub", %{os: :linux})
```

Comment:
```suggestion
    xdg_home_nh = :filename.basedir(:user_data, "nerves-hub")
```

It seems like we should trust whatever basedir tells us that we should use for storing data so that this does the right thing on OSX. I "think" that renaming the variables to remove the XDG and updating the error text to refer to this as the user data directory below is what's needed to make this generic.

I.e., the same logic can still be used:

1. Use `Application.get_env(:nerves_hub_cli, :home_dir)`
2. Use `$NERVES_HUB_HOME`
3. If `~/.nerves-hub` and not the `:user_data` location, tell the user to move the directory
4. Use the `:user_data` location